File: microservicio-clinica-nest/src/modules/patients/dto/update-patient-in.dto.ts
import { PartialType } from '@nestjs/mapped-types';
import { CreatePatientInDto } from './create-patient-in.dto';
import { IsEnum, IsOptional, IsString, IsDateString } from 'class-validator';
import { ApiPropertyOptional } from '@nestjs/swagger';
import { Gender, PatientStatus } from '../entities/patient.entity';

export class UpdatePatientInDto extends PartialType(CreatePatientInDto) {
  @ApiPropertyOptional({ 
    example: 'Juan',
    description: 'Nombre(s) del paciente',
  })
  @IsString({ message: 'El nombre debe ser texto' })
  @IsOptional()
  firstName?: string;

  @ApiPropertyOptional({ 
    example: 'Pérez García',
    description: 'Apellidos del paciente',
  })
  @IsString({ message: 'Los apellidos deben ser texto' })
  @IsOptional()
  lastName?: string;

  @ApiPropertyOptional({ 
    example: '1985-05-15',
    description: 'Fecha de nacimiento en formato YYYY-MM-DD',
  })
  @IsDateString({}, { message: 'La fecha debe tener formato YYYY-MM-DD' })
  @IsOptional()
  birthDate?: string;

  @ApiPropertyOptional({ 
    enum: Gender,
    example: Gender.MASCULINO,
    description: 'Género del paciente',
  })
  @IsEnum(Gender, { message: 'El género debe ser Masculino, Femenino u Otro' })
  @IsOptional()
  gender?: Gender;

  @ApiPropertyOptional({ 
    enum: PatientStatus,
    example: PatientStatus.SEGUIMIENTO,
    description: 'Estado clínico del paciente',
  })
  @IsEnum(PatientStatus, { message: 'El estado debe ser Activo, Seguimiento o Inactivo' })
  @IsOptional()
  status?: PatientStatus;
}

File: autenticación/src/main/java/com/creators/autenticacion/models/dto/patients/UpdatePatientInDto.java
package com.creators.autenticacion.models.dto.patients;

import com.fasterxml.jackson.annotation.JsonInclude;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@Schema(description = "DTO para actualizar un paciente existente")
public class UpdatePatientInDto {

    @Schema(description = "Nombre del paciente", example = "Juan Carlos")
    private String firstName;

    @Schema(description = "Apellido del paciente", example = "Pérez López")
    private String lastName;

    @Schema(description = "Fecha de nacimiento (YYYY-MM-DD)", example = "1990-05-10")
    private String birthDate;

    @Schema(description = "Género del paciente", example = "Masculino", allowableValues = {"Masculino", "Femenino", "Otro"})
    private String gender;

    @Schema(description = "Estado clínico del paciente", example = "Seguimiento", allowableValues = {"Activo", "Seguimiento", "Inactivo"})
    private String status;
}
